src/v/config/configuration.cc

```diff
@@ -353,6 +353,12 @@ configuration::configuration()
       "refreshed",
       required::no,
       2s)
+  , max_connection_creation_rate(
+      *this,
+      "max_connection_creation_rate",
```

Comment:
No, we don't have to be identical.  Since we use `_` and they use `.`, the names are already different, we might as well take advantage of the opportunity to improve the names in other ways.